### ISSUE TITLE
Trouble getting MapClicked event to fire

### ISSUE THREAD

**Issue by DannyDenver** [TorbenK/TK.CustomMap#198]
I got the map to display on the app but when I try to click a location or press on a location no event is fired.

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Xamarin.Forms;
using Xamarin.Forms.Maps;
using Plugin.Geolocator;
using System.Threading;
using TK.CustomMap.Api;
using TK.CustomMap.Interfaces;
using TK.CustomMap;

namespace goD4DMobile
    {
    public partial class Map : ContentPage
        {
        public Map()
            {
            InitializeComponent();

           this.CreateView();

            }

        private async void CreateView()
            {
            
            var newYork = new Position(40.7142700, -74.0059700);
            var mapView =  new TKCustomMap(MapSpan.FromCenterAndRadius(newYork, Distance.FromKilometers(2)));
            mapView.IsShowingUser = true;
            mapView.SetBinding(TKCustomMap.CustomPinsProperty, "Pins");
            mapView.SetBinding(TKCustomMap.MapClickedCommandProperty, "MapClickedCommand");
            mapView.SetBinding(TKCustomMap.MapLongPressCommandProperty, "MapLongPressCommand");
            mapView.SetBinding(TKCustomMap.MapCenterProperty, "MapCenter");
            mapView.SetBinding(TKCustomMap.PinSelectedCommandProperty, "PinSelectedCommand");
            mapView.SetBinding(TKCustomMap.SelectedPinProperty, "SelectedPin");
            mapView.SetBinding(TKCustomMap.RoutesProperty, "Routes");
            mapView.SetBinding(TKCustomMap.PinDragEndCommandProperty, "DragEndCommand");
            mapView.SetBinding(TKCustomMap.CirclesProperty, "Circles");
            mapView.SetBinding(TKCustomMap.CalloutClickedCommandProperty, "CalloutClickedCommand");
            mapView.SetBinding(TKCustomMap.PolylinesProperty, "Lines");
            mapView.SetBinding(TKCustomMap.PolygonsProperty, "Polygons");
            mapView.SetBinding(TKCustomMap.MapRegionProperty, "MapRegion");
            mapView.SetBinding(TKCustomMap.RouteClickedCommandProperty, "RouteClickedCommand");
            mapView.SetBinding(TKCustomMap.RouteCalculationFinishedCommandProperty, "RouteCalculationFinishedCommand");
            mapView.SetBinding(TKCustomMap.TilesUrlOptionsProperty, "TilesUrlOptions");
            mapView.SetBinding(TKCustomMap.MapFunctionsProperty, "MapFunctions");
            mapView.IsRegionChangeAnimated = true;

            mapView.MapClicked += MapView_MapClicked;
            mapView.MapLongPress += MapView_MapLongPress;

            //this.Content = mapView;
            this.Content = mapView;

            }

        private void MapView_MapLongPress(object sender, TKGenericEventArgs<Position> e)
            {
            throw new NotImplementedException();
            }

        private void MapView_MapClicked(object sender, TKGenericEventArgs<Position> e)
            {
            throw new NotImplementedException();
            }

**Comment by PhilJenson** [comment]
On Android I found that you need to also assign the MapClickCommand to get the MapClick event to work. This appears to be a minor bug. I have not tried iOS.

The code below shows the TKCustomMap Andriod render, it ignores any click event if this is not assigned

`        private void OnMapClick(object sender, GoogleMap.MapClickEventArgs e)
        {
            if (this.FormsMap == null || this.FormsMap.MapClickedCommand == null) return;`

**Comment by johnm01** [comment]
Hi - apologies for not being a skygod of this sort of stuff (I have to struggle to learn by copying) but I am having huge difficulty getting mapclicked to fire also. I've taken the tk-custom map sample and inserted a mapclicked delegate - I then  create a new pcl project - copy and paste all the code from the TK custommap sample using the nuget package 1.5.2 pre 1 -  

2 questions   1)  the TKcustommap sample creates its own tkcustommap.dll  so does not need the nuget installed - is the code somehow different with the latest 1.5.2 nuget package 
2) (probably a different issue) when I create a new project and paste the sample code - even though I paste in  FormsMaps.Init(("proper key")  I still get a message saying MapService token not specified
- in other words - works fine on hte TKCustommap sample even when I adapt it to my own key but when I create a new project install the nuget package and paste in same code then I get 2 failures - one the mapclicked event doesn't fire and 2 ) it reports no map token
Is there an issue with 1.5.2 pre
At present I'm testing on a PCL project with only UWP

**Comment by johnm01** [comment]
A follow on to last. I have the mapclicked event firing in the TKSample pclproject BUT only when running on VS2017.(previously running on VS2015  I thought at first this was because VS2017 allows MS.NetCore.UWP ver 5.3 but No,  the project from VS2015 still has .netcore.UWP 5.1 when imported into VS2017.
I also thought because I couldn't get a simple xamforms pcl project with TKcustommap nuget installed running (wellno of course I can get it running -its just the mapclicked wont fire) I thought to go back to v 1.5.1 but that made no difference.
 My code is the same as posted above. the only difference is that in the TKsample sln because it does not rely on the nuget package and has multiple models it sets the binding context thus 
   this.BindingContext = new SampleViewModel();
So I've included samplemodel in my new copy/paste pcl project but still no dice  YET the mapclicked event fires when running the original TKCustommap sample in VS2017


I've tried putting in a MapClickedCommand (as suggested above as per Android) but no result
Also the MapLongpress does not fire

I've debugged inch by inch but cannot find what is being called differently in the two projects. The only noticeable difference is (as I said above) I keep getting the Maptoken warning despite having a valid Formsmaps.init(Token)  - which works in the TKCustomsample

I'd appreciate any help - my modus operandi of copy/deducing/mimicking is slow work 
Would any kind soul like to look at the 2 projects and see what I'm missing?
Could anyone supply a simple PCL solution using the  nuget package which demonstrates JUST the mapclicked event and not a lot of confusing other stuff - so difficult to see what depends on what

**Comment by johnm01** [comment]
Cracked IT !!    nothing to do with tokens nothing to do with versions 
teh Nuget install even Vers 1.5.2 does not install TKCustommap.UWP 
I copied this dll fromthe TKcustomMap sample project and everything worked 

Now going to work through the Android and Ios 

Is the nuget package supposed to install the UWP.dll????
